### ISSUE TITLE
[web] Autocreate settings page

### ISSUE THREAD

**Issue by ejurgensen** [owntone/owntone-server#1400]
Currently, when adding a new setting (via settings.c), it isn't immediately available to use (except via the api) before the web interface settings page is updated to display it. I was thinking if it would be possible that the page automatically displayed the current set of settings?

I realise there could be some design challenges with this, so perhaps it's better I just learn how to modify the web UI. What are your thoughts, @chme?

**Comment by chme** [comment]
Yes, I agree, that getting the settings automatically in the UI would be good.
And yes, that can be implemented. We will have to expose additional information for each setting, to make it look decent in the UI. I will look into it.

**Comment by ma-ku** [comment]
Just an idea; look at the HomeBridge project. They support custom settings via JSON Schema files. There is a library that supports rendering settings from the JSON definition. Maybe a shortcut?
